### PR TITLE
Add Error handling and redirecting

### PR DESCRIPTION
**Pull request by adodge-bcg**
# Description

This PR includes the following proposed change(s):

- update demo app error handling to receive as queryParams
- update frontend error catch blocks to redirect with status and msg

Addresses [FLA-222](https://justice.gov.bc.ca/jira/browse/FLA-222)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/efiling-demo/src/components/page/error/Error.js [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+import { useLocation } from "react-router-dom";
 import { Header, Footer, Button } from "shared-components";
 import { propTypes } from "../../../types/propTypes";
 
 export default function Error({ page: { header, error } }) {
+  const location = useLocation();
+  const queryParams = queryString.parse(location.search);
+
+  const status = queryParams.status;
+  const message = queryParams.message;
```
Comment (sdevalapurkar-bcgov) [inline src/frontend/efiling-demo/src/components/page/error/Error.js:13]: the way this is being done needs to be discussed, the way I had set it up earlier is that the querystring gets parsed at a HOC and then passed in as a prop to this component.

If you'd like to use it this way, then we should remove the code in the HOC to parse the error and pass it into this component

Comment (adodge-bcg) [reply src/frontend/efiling-demo/src/components/page/error/Error.js:13]: No you right, I missed what it was doing there when I was looking at it yesterday.

File: src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.test.js [hunk 2]
```diff
@@ -56,4 +58,25 @@ describe("CSOAccount Component", () => {
...
+    );
+
+    expect(getByText(container, "Create CSO Account").disabled).toBeTruthy();
+    fireEvent.click(getByRole(container, "checkbox"));
+    expect(getByText(container, "Create CSO Account").disabled).toBeFalsy();
+
+    fireEvent.click(getByText(container, "Create CSO Account"));
+    await wait(() => {});
+
+    expect(window.open).toHaveBeenCalled();
```
Comment (sdevalapurkar-bcgov) [inline src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.test.js:80]: can we assert `toHaveBeenCalledWith("url", "_self")`? its a stronger assertion

File: src/frontend/efiling-frontend/src/components/page/home/Home.test.js [hunk 3]
```diff
@@ -71,14 +73,16 @@ describe("Home", () => {
...
+    mock.onGet(apiRequest).reply(400, { message: "There was an error" });
 
     const { asFragment } = render(component);
 
     await wait(() => {
       expect(asFragment()).toMatchSnapshot();
       expect(sessionStorage.getItem("cancelUrl")).toBeFalsy();
     });
+
+    expect(window.open).toHaveBeenCalled();
```
Comment (sdevalapurkar-bcgov) [inline src/frontend/efiling-frontend/src/components/page/home/Home.test.js:85]: same here

File: src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.test.js (hunk 2 above)
Comment (adodge-bcg) [reply src/frontend/efiling-frontend/src/components/page/cso-account/CSOAccount.test.js:80]: fixed

File: src/frontend/efiling-frontend/src/components/page/home/Home.test.js (hunk 3 above)
Comment (adodge-bcg) [reply src/frontend/efiling-frontend/src/components/page/home/Home.test.js:85]: fixed

**Comment by sdevalapurkar-bcgov** [comment]
looks like some tests are failing, will take another look once you've addressed those but should be good to merge once theyre fixed I think

**Pull request merged**